tests/nightly/test_np_large_array.py

```diff
@@ -568,6 +502,517 @@ def test_slice_assign():
     B[-1] = 2
     assert B[-1, 0] == 2 and B[-1, 1] == 2
 
+@use_np
+def test_logical_family():
+    def batch_check(x1, x2, funcs):
+        x1.attach_grad()
+        for f in funcs:
+            with mx.autograd.record():
+                y = f(x1, x2)
+                y.backward()
+            assert y.shape == x1.shape
+            assert y[0] == f(x1[0], x2[0])
+            assert x1.grad.shape == x1.shape
+            assert x1.grad[0] == 0
+
+    A = np.zeros((INT_OVERFLOW), dtype='int32')
+    B = np.ones((INT_OVERFLOW), dtype='int32')
+    batch_check(A, B, [np.logical_and, np.logical_or, np.logical_xor])
+    B.attach_grad()
+    with mx.autograd.record():
+        C = np.logical_not(B)
+        C.backward()
+    assert C.shape == B.shape
+    assert C[0] == 0
+    assert B.grad.shape == B.shape
+    assert B.grad[0] == 0
+
+@use_np
+def test_deg_rad():
+    # deg2rad is the same thing as radians
+    # rad2deg is the same thing as degrees
+    A = np.zeros((INT_OVERFLOW, 2))
+    A[-1, -1] = 180
+    A.attach_grad()
+    with mx.autograd.record():
+        B = np.deg2rad(A)
+        B.backward()
+    assert B.shape == A.shape
+    assert B[0, 0] == 0
+    assert_almost_equal(B[-1, -1], np.array([np.pi]), rtol=1e-5, atol=1e-5)
+    assert A.grad.shape == A.shape
+    assert_almost_equal(A.grad[0, 0], np.array([1.0 / 180 * np.pi]), rtol=1e-5, atol=1e-5)
+    B.attach_grad()
+    with mx.autograd.record():
+        C = np.rad2deg(B)
+        C.backward()
+    assert C.shape == B.shape
+    assert C[0, 0] == 0 and C[-1, -1] == 180
+    assert B.grad.shape == B.shape
+    assert_almost_equal(B.grad[0, 0], np.array([180.0 / np.pi]), rtol=1e-5, atol=1e-5)
+
+@use_np
+def test_divide():
+    # np.divide and np.true_divide are the same thing
+    A = np.ones((INT_OVERFLOW, 2))
+    A[-1, -1] = 10
+    A.attach_grad()
+    with mx.autograd.record():
+        B = np.divide(A, np.array([2, 3]))
+        B.backward()
+    assert B.shape == A.shape
+    assert_almost_equal(B[-1, -1], np.array([10 / 3]), rtol=1e-5, atol=1e-5)
+    assert A.grad.shape == A.shape
+    assert_almost_equal(A.grad[-1, -1], np.array([1.0 / 3]), rtol=1e-5, atol=1e-5)
+
+@use_np
+def test_minimum():
+    A = np.ones((INT_OVERFLOW, 2))
+    A[-1, -1] = -1
+    B = np.zeros((INT_OVERFLOW, 1))
+    A.attach_grad()
+    B.attach_grad()
+    with mx.autograd.record():
+        C = np.minimum(A, B)
+        C.backward()
+    assert C.shape == A.shape
+    assert C[-1, -1] == -1
+    assert A.grad.shape == A.shape
+    assert A.grad[-1, -1] == 1 and A.grad[0, 0] == 0
+    assert B.grad.shape == B.shape
+    assert B.grad[-1] == 1 and B.grad[0] == 2
```

Comment:
why B.grad[0] and B.grad[-1] are different??

Reply:
Here B is broadcast to have match the shape of A. B shape (X, 1) -> (X, 2) so the gradients are also effectively doubled